spec/basic/elements_spec.js

```diff
@@ -1,466 +1,473 @@
+    expect(await usernameInput.equals(usernameInput)).toEqual(true);
+    expect(await usernameInput.equals(name)).toEqual(false);
   });
 });
 
-describe('ElementArrayFinder', function() {
+describe('ElementArrayFinder', () => {
 
-  it('action should act on all elements', function() {
-    browser.get('index.html#/conflict');
+  it('action should act on all elements', async() => {
+    await browser.get('index.html#/conflict');
 
-    var multiElement = element.all(by.binding('item.reusedBinding'));
-    expect(multiElement.getText()).toEqual(['Outer: outer', 'Inner: inner']);
+    const multiElement = element.all(by.binding('item.reusedBinding'));
+    expect(await multiElement.getText())
+        .toEqual(['Outer: outer', 'Inner: inner']);
   });
 
-  it('click action should act on all elements', function() {
-    var checkboxesElms = $$('#checkboxes input');
-    browser.get('index.html');
+  it('click action should act on all elements', async() => {
+    const checkboxesElms = $$('#checkboxes input');
+    await browser.get('index.html');
 
-    expect(checkboxesElms.isSelected()).toEqual([true, false, false, false]);
-    checkboxesElms.click();
-    expect(checkboxesElms.isSelected()).toEqual([false, true, true, true]);
+    expect(await checkboxesElms.isSelected())
+        .toEqual([true, false, false, false]);
+    await checkboxesElms.click();
+    expect(await checkboxesElms.isSelected())
+        .toEqual([false, true, true, true]);
   });
 
-  it('action should act on all elements selected by filter', function() {
-    browser.get('index.html');
+  it('action should act on all elements selected by filter', async() => {
+    await browser.get('index.html');
 
-    var multiElement = $$('#checkboxes input').filter(function(elem, index) {
+    const multiElement = $$('#checkboxes input').filter((_, index) => {
       return index == 2 || index == 3;
     });
-    multiElement.click();
-    expect($('#letterlist').getText()).toEqual('wx');
+    await multiElement.click();
+    expect(await $('#letterlist').getText()).toEqual('wx');
   });
 
-  it('filter should chain with index correctly', function() {
-    browser.get('index.html');
+  it('filter should chain with index correctly', async() => {
+    await browser.get('index.html');
 
-    var elem = $$('#checkboxes input').filter(function(elem, index) {
+    const elem = $$('#checkboxes input').filter((_, index) => {
       return index == 2 || index == 3;
     }).last();
-    elem.click();
-    expect($('#letterlist').getText()).toEqual('x');
+    await elem.click();
+    expect(await $('#letterlist').getText()).toEqual('x');
   });
 
-  it('filter should work in page object', function() {
-    var elements = element.all(by.css('#animals ul li')).filter(function(elem) {
-      return elem.getText().then(function(text) {
-        return text === 'big dog';
-      });
+  it('filter should work in page object', async() => {
+    const elements = element.all(by.css('#animals ul li'))
```

Comment:
I'm not sure, probably before `element.all` should be `await` keyword because of using `filter()`

Reply:
done.